### PR TITLE
Issue 2840050: Use Dependency Injection for LocalStockService

### PR DESCRIPTION
**Pull request by olafkarsten**
Makes the LocalStockChecker and LocalStockUpdater a service an injects them in the LocalStockService. Adds test for the LocalStockService.

### CODE REVIEW

**Comment by BBGuy** [comment]
Hi @olafkarsten looks like me pushing configuration injection https://github.com/BBGuy/commerce_stock/pull/5 broke this. Can you fix?

File: modules/local_storage/commerce_stock_local.services.yml [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
+  commerce_stock.local_stock_storage:
+     class: Drupal\commerce_stock_local\LocalStockStorage
+     arguments: ['@entity_type.manager', '@database']
+     public: false
+
+  commerce_stock.local_stock_checker:
+    alias: commerce_stock.local_stock_storage
+
+  commerce_stock.local_stock_updater:
+     alias: commerce_stock.local_stock_storage
```
Comment (steveoliver) [inline modules/local_storage/commerce_stock_local.services.yml]: Indent by 2, not 3.

File: modules/local_storage/commerce_stock_local.services.yml [hunk 2]
```diff
@@ -1,5 +1,17 @@
 services:
   commerce_stock.local_stock_service:
     class: Drupal\commerce_stock_local\LocalStockService
+    arguments: ['@commerce_stock.local_stock_checker', '@commerce_stock.local_stock_updater']
     tags:
       - { name: commerce_stock.stock_service, priority: 0 }
+
+  commerce_stock.local_stock_storage:
+     class: Drupal\commerce_stock_local\LocalStockStorage
```
Comment (steveoliver) [inline modules/local_storage/commerce_stock_local.services.yml]: Indent these by 2, not 3.

File: modules/local_storage/src/LocalStockService.php [hunk 3]
```diff
@@ -2,9 +2,17 @@
...
 use Drupal\commerce_stock\StockServiceInterface;
 use Drupal\commerce_stock\StockServiceConfig;
+use Drupal\commerce_stock\StockUpdateInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 
+/**
+ * Class LocalStockService.
+ *
+ * @package Drupal\commerce_stock_local
+ */
```
Comment (steveoliver) [inline modules/local_storage/src/LocalStockService.php]: Class docblock not needed.

File: modules/local_storage/src/LocalStockService.php [hunk 4]
```diff
@@ -29,16 +37,34 @@ class LocalStockService implements StockServiceInterface {
   protected $stockServiceConfig;
 
   /**
-   * Constructor for the service.
+   * LocalStockService constructor.
+   *
+   * @param StockCheckInterface $stock_checker
+   *   The stock checker.
+   * @param StockUpdateInterface $stock_updater
+   *    The stock updater.
```
Comment (steveoliver) [inline modules/local_storage/src/LocalStockService.php]: Indentation.

File: modules/local_storage/src/LocalStockService.php [hunk 5]
```diff
@@ -29,16 +37,34 @@ class LocalStockService implements StockServiceInterface {
   protected $stockServiceConfig;
 
   /**
-   * Constructor for the service.
+   * LocalStockService constructor.
```
Comment (steveoliver) [inline modules/local_storage/src/LocalStockService.php]: "Constructs a new LocalStockService object."

File: modules/local_storage/src/LocalStockService.php [hunk 6]
```diff
@@ -29,16 +37,34 @@ class LocalStockService implements StockServiceInterface {
...
+   *
+   * @param StockCheckInterface $stock_checker
+   *   The stock checker.
+   * @param StockUpdateInterface $stock_updater
+   *    The stock updater.
    */
-  public function __construct() {
+  public function __construct(StockCheckInterface $stock_checker, StockUpdateInterface $stock_updater) {
     // Create the objects needed.
```
Comment (steveoliver) [inline modules/local_storage/src/LocalStockService.php]: Can remove this comment

File: tests/src/Kernel/CommerceStockKernelTestBase.php [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace Drupal\Tests\commerce_stock\Kernel;
+
+use Drupal\Tests\commerce\Kernel\CommerceKernelTestBase;
+
+/**
+ * Class CommerceStockKernelTestBase.
+ *
+ * Provides a base class for commerce stock and submodules.
+ */
```
Comment (steveoliver) [inline tests/src/Kernel/CommerceStockKernelTestBase.php:9]: Class docblock can be just one line: "Base class for Commerce Stock kernel tests."

File: modules/local_storage/tests/src/Kernel/LocalStockServiceTest.php [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    $prophecy->getLocationList(Argument::any())->willReturn([1 => 'main']);
+    $stockChecker = $prophecy->reveal();
+    $stockUpdater = $this->prophesize(StockUpdateInterface::class)->reveal();
+    $localStockService = new LocalStockService($stockChecker, $stockUpdater);
+    self::assertEquals($stockChecker, $localStockService->getStockChecker());
+    self::assertEquals($stockUpdater, $localStockService->getStockUpdater());
+    self::assertEquals('local_stock', $localStockService->getId());
+    self::assertEquals('Local stock', $localStockService->getName());
+
+    // Test that instation through container works.
```
Comment (steveoliver) [inline modules/local_storage/tests/src/Kernel/LocalStockServiceTest.php]: spelling: instantiation

File: modules/local_storage/tests/src/Kernel/LocalStockServiceTest.php [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+use Drupal\commerce_stock\StockCheckInterface;
+use Drupal\commerce_stock\StockServiceConfigInterface;
+use Drupal\commerce_stock\StockUpdateInterface;
+use Drupal\commerce_stock_local\LocalStockService;
+use Drupal\Tests\commerce_stock\Kernel\CommerceStockKernelTestBase;
+use Prophecy\Argument;
+
+/**
+ * Tests the LocalStockService.
```
Comment (steveoliver) [inline modules/local_storage/tests/src/Kernel/LocalStockServiceTest.php]: "Tests the local stock service."

File: modules/local_storage/src/LocalStockService.php [hunk 10]
```diff
@@ -29,16 +37,34 @@ class LocalStockService implements StockServiceInterface {
...
-    $this->stockChecker = new LocalStockStorage();
-    $this->stockUpdater = $this->stockChecker;
+    $this->stockChecker = $stock_checker;
+    $this->stockUpdater = $stock_updater;
     $this->stockServiceConfig = new StockServiceConfig($this->stockChecker);
   }
 
   /**
+   * @param ContainerInterface $container
+   *    The DI container.
```
Comment (steveoliver) [inline modules/local_storage/src/LocalStockService.php]: Indentation.

**Review by steveoliver — CHANGES_REQUESTED** [review]
Looks good -- just a few nitpicks.

**Comment by steveoliver** [comment]
As I commented in https://www.drupal.org/node/2840050, let's use this PR to split LocalStockStorage into LocalStockChecker and LocalStockUpdater as well as the DI work you've already done.  Then we don't need the aliases.

**Comment by BBGuy** [comment]
committed to gh and d.o https://www.drupal.org/commitlog/commit/21250/f5a447d7f0468fb144ca0bc25b49b051c6ce970b

**Pull request closed without merging**